src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1658,6 +1668,20 @@ export class SavedObjectsRepository {
    *   .incrementCounter('dashboard_counter_type', 'counter_id', [
    *     'stats.apiCalls',
    *   ])
+   *
+   * // Increment the apiCalls field counter by 4
+   * repository
+   *   .incrementCounter('dashboard_counter_type', 'counter_id', [
+   *     { fieldName: 'stats.apiCalls' incrementBy: 4 },
+   *   ])
+   *
+   * // Initialize the document with arbitrary fields if not present
+   * repository.incrementCounter<{ appId: string }>(
+   *   'dashboard_counter_type',
+   *   'counter_id',
+   *   [ 'stats.apiCalls'],
+   *   { upsertAttributes: { appId: 'myId' } }
```

Comment:
> Should upsertAttributes?: T; be upsertAttributes?: Partial<T>

The example is not really precise, but the `upsertAttributes ` should contains all the **required** properties of the SO type, to avoid inserting corrupted objects. See https://github.com/elastic/kibana/issues/90922#issuecomment-778135871 for more details